src/lib/promote/Promote.ts

```diff
@@ -169,9 +170,10 @@ export class Promote {
     }
 
     await Promise.all(promises)
-    if (await buildVersions.saveAsync()) {
-      await buildVersions.gitRepo.commitAndPushAsync('update build versions')
-    }
+    if (await buildVersions.saveAsync(commitMessage)) {
+      await buildVersions.gitRepo.commitAndPushAsync(commitMessage)
```

Comment:
I think you may have discovered a redundant line of code here, since `buildVersions.saveAsync` already does call `this.gitRepo.commitAndPushAsync(message)`

Reply:
Yeah I was actually hung up for a bit on that cause I had only been changing `buildVersions.gitRepo.commitAndPushAsync(commitMessage)`. I missed that saveAsync actually had the value set in the argument. I can remove that and test it out if you want?

Reply:
Yea if you could remove the redundancy that would be great!

Reply:
For sure! It looks like BuildVersions saveAsync function is only called in this instance for promote:complete-publish and build:complete-publish as well so should I just delete it all together? Will just need to call commitAndPushAsync in its use in the tests in /test/git/git.test.ts.

Reply:
Wait nevermind just have to remove the commit and push async call.